### DISCUSSION TITLE
题图

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#2487] [插件说明]
描述
---

> 为你的阅读模式提供 **题图**。

地址
---

[https://simpread.ksria.cn/plugins/details/PcmPCT9rgM](https://simpread.ksria.cn/plugins/details/PcmPCT9rgM)

效果
---

可以 [模仿 Safari 阅读模式样式](https://simpread.ksria.cn/plugins/details/tbjTx33iAA) 配合使用。

![image](https://github.com/Kenshin/simpread/assets/81074/616da56d-ebb0-4638-831e-b664376496e1)

选项
---

随便进入一个页面 e.g. https://sspai.com/post/71576 并进入 **阅读模式 → 右下角 → 动作 → 插件触发器**

按照下图所示即可查看此插件的选项页。（注意：如果你的界面没有插件触发器，请根据此 [解决方案](https://github.com/Kenshin/simpread/discussions/2342) 修复。

<img width="1086" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/7819d9f4-e55a-48d5-8e9f-3b3f4b156251">

功能
---

1. 设置题图来源，包括：`当前页的题图` `正文的第一张图片` `Unsplash 来源`

    ![image](https://user-images.githubusercontent.com/81074/183030637-79b33ea6-5ec7-4b06-bfef-42490ca61c09.png)

2. 一般使用题图建议安装 [模仿 Safari 阅读模式样式](https://simpread.ksria.cn/plugins/details/tbjTx33iAA) 后，可得到（最上面第一个图的效果）

    <img width="468" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/6556b776-10f5-4ba3-af44-f4d703cd778e">

3. 支持使用 Unsplash 做作为题图，并支持 [定制化](https://github.com/Kenshin/simpread/discussions/2487#discussioncomment-3330284) 

     <img width="478" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/5d57b004-e15e-40be-8a53-a4653efdf292">

4. 当题图的宽度小于布局时，是否使用 Unsplash 作为背景源

     <img width="463" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/5aa2d765-f0a9-4dd3-a263-36702c6869bc">

5. 白名单机制

    ![image](https://user-images.githubusercontent.com/81074/183027221-719e51ee-86ce-488b-9d72-df39daf5e375.png)

注意
---

因 [Unsplash Source 彻底下线](https://changelog.unsplash.com/deprecations/2021/11/25/source-deprecation.html) 了，所以当你在使用题图时遇到了下图所示的效果，请升级到 1.1.0（含），即可自动解决此问题。

![image](https://github.com/Kenshin/simpread/assets/81074/0e3f12d6-6c50-4812-8873-56ba647d3e5f)

更新日志
---

### 1.2.0

- `Fix` 访问 utgd.net 因没有按照通用结构设置导致无法获取题图的问题 https://github.com/Kenshin/simpread/issues/7062

- `Add` 白名单机制 jQuery 字符串方案 e.g.  `{"site":"utgd.net","query":"{{$( 'head meta[name="ogimage"]' ).attr( 'content' )}}"}` 详细说明 https://github.com/Kenshin/simpread/discussions/2487#discussioncomment-3330699

### 1.1.1

- `Fix` 当题图小于宽度时不显示题图的错误 [#5797](https://github.com/Kenshin/simpread/issues/5797)

### 1.1.0

- `Add` 更换 `Unsplash Source` 改为 `Unsplash API`

- `Add` 定制 `Query` 参数

- `Add` 定制 `Size` 参数

- `Optimize` 去掉 *`un_source` 相关逻辑*

- `Optimize` 题图的设置逻辑，依次为：白名单、Unsplash、题图、正文第一个图

- `Optimize` 更改 `白名单` 为专门用于定制的参数

### 1.0.5

- `Add` [白名单机制](https://github.com/Kenshin/simpread/discussions/2487#discussioncomment-3330699)

- `Add` [占位符](https://github.com/Kenshin/simpread/discussions/2487#discussioncomment-3331231)

### 1.0.4

- `Fix` 某些页面样式错误问题

- `Add` 当题图的宽度小于布局时，是否使用 Unsplash 来源？

- `Add` 题图来源设置为：`当前页的题图` `正文的第一张图片` `Unsplash 来源`

### 1.0.3

> 版本号初始变更为 1.0.0

- 当没有勾选  [unsplash.com](http://unsplash.com) 并且也没有题图时，[Title 无法显示](https://happyxiao.com/headline/)。

- [unsplash.com](http://unsplash.com) 等完成加载后再删除标题

### 1.0.2（原 0.0.2）

- 修复加入稍后读后多个 `sr-rd-title` 的问题

- favicon [size 错误](https://nobyda.github.io/2021/06/08/Surge_network_firewall/#%E6%96%B9%E6%A1%88%E4%B8%80-%E6%B3%A8%E6%84%8F%E4%BA%8B%E9%A1%B9)

### 1.0.0（原 0.0.1）

1. 自动获取当前页面的题图

2. 当没有题图时，可选：

    - 是否将第一个图片作为题图？

    - 是否使用 [Unsplash Source URL](https://source.unsplash.com/) 作为题图？

引申
---

> 安装插件或带有插件触发器的插件失效的 [解决方案](https://github.com/Kenshin/simpread/discussions/2342)。

**Comment by OptimusPrimeme** [comment]
[Unsplash 用新的 API 了](https://changelog.unsplash.com/deprecations/2021/11/25/source-deprecation.html)，不过旧的还能继续用，😊

**Reply by Kenshin** [reply]
哈哈，谢谢提示。

好久没看到你了~

**Comment by Kenshin** [comment]
# Unsplash 设置

## 注意

题图 1.1.0 版升级了 Unsplash Source 为 Unsplash API，所以下面的内容以 Unsplash API 为基准说明，如果你的题图选项界面与下面说明不一样，请升级题图插件。

## Unsplash API

当开启下图所示的选项后，当没有题图时，会自动获取 Unsplash 作为题图。

<img width="479" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/3a6af340-2996-4862-9552-9b4f17355db6">

## 默认值

用户继续开启这个选项，即便什么都不做就可以实现将 Unsplash 作为题图。

## 定制化

### 尺寸

值可不设置，也可以为 `1402x708` 或仅设置宽度 `1400` 同时宽度支持 [占位符](https://github.com/Kenshin/simpread/discussions/2487#discussioncomment-3331231)

<img width="473" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/44ec447b-9f87-49e5-a225-ae963c330813">

### 检索

当不设置时为 `topics=wallpapers&orientation=landscape&order_by=popular`

<img width="473" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/e110b14a-e522-41d1-aea7-76135da9e490">

支持定制，细节 [请看这里](https://unsplash.com/documentation#get-a-random-photo)。

<img width="732" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/6d2e3e50-548e-4fea-9176-a99f90b08767">

**Reply by Kenshin** [reply]
# 占位符

1.0.5 版增加了占位符，共有三种：

|                  | 说明     |
|------------------|--------|
| `{{width}}`        | 正文的宽度  |
| `{{screen_width}}` | 分辨率的宽度 |
| `{{win_width}}`    | 浏览器的宽度 |

## 用法

- 不设置

- 定值 `1400x540` 或 `{{width}}x540`

- 仅设置宽度 `1400`  或 `{{screen_width}}`

**Comment by Kenshin** [comment]
# 白名单

白名单是题图 1.0.5 增加的功能，可以更精细的设定题图。

## 注意

题图 1.1.0 版重构了白名单机制，支持 `query` 参数。

## 结构

```
{"site":"site", "pic":"<尺寸>", "query":"<查询>"}
```

|       | 说明          | 例子                                               | 说明                   |
|-------|-------------|--------------------------------------------------|----------------------|
| site  | 支持 域名 与 URL | `twitter.com` 或 `https://sspai.com/post/69972`   | 必填，为 `-1` 时当前规则不设置题图 |
| pic   | 题图的尺寸       | `{{screen_width}}x540` `1400x480` `{{width}}`    | 可为空，即使用默认尺寸规则        |
| query | 题图的查询规则     | `topics=textures-patterns&orientation=landscape` | 可为空，即使用默认查询规则        |
| query | 题图的查询规则     | `{{xxxx}}` | 可为空，支持 jQuery 字符执行方式        |

### 注意

1.2.0 版 query 增加了 jQuery 字符串执行方式，由 `{{` 和 `}}` 作为标识，例如：

-  `{{$( 'head meta[name="ogimage"]' ).attr( 'content' )}}`

- `{{(()=>{return $( 'head meta[name="ogimage"]'.attr('content')})()}}`

## 不设置题图

当 `pic` 的值为 `-1` 时，不设置当前页面的题图。e.g. `{"site":"https://sspai.com/post/78480","pic":"-1"}`

即当访问 `https://sspai.com/post/78480` 不设置题图。

## 几个例子

```
{"site":"twitter.com","pic":"{{screen_width}}x540"}
{"site":"yuque.com","pic":"{{win_width}}"}
{"site":"chat.openai.com"}
{"site":"https://sspai.com/post/78480","pic":"-1"}
{"site":"https://simone.org/tracking-screen-time/","pic":"1400x400","query":"topics=textures-patterns&orientation=landscape"}
{"site":"utgd.net","query":"{{$( `head meta[name=\"ogimage\"]` ).attr( 'content' )}}"}
```

简单的解释

1. 当主域名为 `twitter.com` 时，设置题图尺寸 `{{screen_width}}x540` 并使用 **默认的查询规则** 得到题图。

2. 当域名为 `https://simone.org/tracking-screen-time/` ，设置题图尺寸 `1400x400` 题图检索规则 `topics=textures-patterns&orientation=landscape`

3. 当主域名为 `chat.openai.com` 时，使用 **默认的题图尺寸与查询规则** 得到题图。

4. 当主域名为 `utgd.net` 时，执行 query 对应的 jQuery 语句，即： `$( 'head meta[name="ogimage"]' ).attr( 'content' )`

## 优先级

1.  白名单

2. 选择 Unsplash

    <img width="469" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/d8193c14-be51-4f20-b514-7fe6508b2469">

3. 当前页的题图

    <img width="471" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/a066bd19-d271-4d94-b4e1-8775f16e88ef">

4. 当前页正文的第一张图片

    <img width="461" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/75db5489-370e-4f22-b046-3b5c006fd923">

**Comment by iblogc** [comment]
讨论内容转移到 https://github.com/Kenshin/simpread/issues/4602

**Comment by MrLeo** [comment]
讨论内容转移到 https://github.com/Kenshin/simpread/issues/5893
